crates/runtime/src/pallets.rs

```diff
@@ -1,9 +1,16 @@
 //! Configuration of the pallets used in the runtime.
 //! The pallets used in the runtime are configured here.
 //! This file is used to generate the `construct_runtime!` macro.
+#[cfg(feature = "dev")]
```

Comment:
But you declared the `dev = []` feature in the `node/Cargo.toml`, not in this crate.

I think you have to add `dev = []` to `runtime/Cargo.toml` and change the `node/Cargo.toml` to `dev = [madara-runtime/dev]`

Reply:
Otherwise this will always be ignored